moveit_ros/moveit_servo/test/pose_tracking_test.cpp

```diff
@@ -135,15 +176,15 @@ TEST_F(PoseTrackingFixture, OutgoingMsgTest)
     size_t msg_count = 0;
     while (++msg_count < 100)
     {
-      target_pose_pub_.publish(target_pose);
-      ros::Duration(0.01).sleep();
+      target_pose_pub_->publish(target_pose);
+      std::this_thread::sleep_for(std::chrono::milliseconds(10));
+      ;
     }
   });
 
-  ros::Duration(ROS_PUB_SUB_DELAY).sleep();
+  std::this_thread::sleep_for(std::chrono::seconds(ROS_PUB_SUB_DELAY));
```

Comment:
This sleep should be removed.

Reply:
Poll for subscription count? https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1PublisherBase.html#a60c84260b2ff0fb364fad688ed91234f

Reply:
Maybe that would work here.  In the ros1 tests for servo we count messages produced by servo on its output for this.  This is the one I'm not sure what the right approach is (as I'm not sure why this sleep is here at all).  This sleep is after the loop that publishes command messages.

Reply:
Looking at the test logic I don't think we should have a delay here at all. The subscription callback should be synchronized or timed out using a future (preferred imo) or condition variable.

Reply:
I removed the sleep because otherwise the difference between header timestamp and current time was larger than the timeout which caused the program to think the target pose was not updating